types/mempool/sender_nonce_test.go

```diff
@@ -34,7 +35,12 @@ func (s *MempoolTestSuite) TestTxOrder() {
 				{p: 15, n: 1, a: sb},
 				{p: 20, n: 1, a: sa},
 			},
-			order: []int{3, 4, 2, 1, 0},
+			order: [][]int{
+				{4, 2, 3, 1, 0},
+				{3, 4, 2, 1, 0},
```

Comment:
Why are there 2 orders specified for each test case?

Reply:
Because the iteration over sender map is not guaranteed to be ordered so Sa or sb can be either index 0 or index 1 in the array interchangeably.

Reply:
Personally I think we should have randomness derived from seed only, not built-in map non-determinism in iteration.